prov/gni/configure.m4

```diff
@@ -26,22 +26,30 @@ AC_DEFUN([FI_GNI_CONFIGURE],[
                                       [Enable xpmem (gni provider) @<:@default=yes@:>@])],
                       )
 
+        AC_ARG_ENABLE([ugni-static],
+                      [AS_HELP_STRING([--enable-ugni-static],
+                                      [Enable static linking with uGNI.  Recommended for KNL.])],
+                     )
+
         AS_IF([test x"$enable_gni" != x"no"],
                [FI_PKG_CHECK_MODULES([CRAY_GNI_HEADERS], [cray-gni-headers],
                                  [gni_header_happy=1
                                   gni_CPPFLAGS="$CRAY_GNI_HEADERS_CFLAGS $gni_CPPFLAGS"
-                                  gni_LDFLAGS="$CRAY_GNI_HEADER_LIBS $gni_LDFLAGS"
                                  ],
                                  [gni_header_happy=0])
               ])
 
         AS_IF([test "$gni_header_happy" -eq 1],
-              [FI_PKG_CHECK_MODULES([CRAY_UGNI], [cray-ugni],
+              [FI_PKG_CHECK_MODULES_STATIC([CRAY_UGNI], [cray-ugni],
```

Comment:
yes, its a customized version of the ```PKG_CHECK_MODULES``` that sets a BLAH_BLAH_STATIC_LIBS shell variable that can be used to pick up any extra dependencies that may be needed if the package is to be linked in statically.  Here we don't actually need it since libugni doesn't have any additional explicit dependencies when being linked statically.